cli/admin/src/main/java/io/pravega/cli/admin/readerGroup/ParseReaderGroupStreamCommand.java

```diff
@@ -0,0 +1,173 @@
+    private final static int TYPE = WireCommandType.EVENT.getCode();
+    private final static int REQUEST_TIMEOUT_SECONDS = 10;
+
+    private final GrpcAuthHelper authHelper;
+
+    public ParseReaderGroupStreamCommand(CommandArgs args) {
+        super(args);
+
+        authHelper = new GrpcAuthHelper(true, "secret", 600);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(4);
+        final String scope = getArg(0);
+        final String readerGroup = getArg(1);
+        final String segmentStoreHost = getArg(2);
+        final String fileName = getArg(3);
+        String stream = NameUtils.getStreamForReaderGroup(readerGroup);
+
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient);
+        String tmpfilename = "tmp/RG_" + readerGroup + "_" + new Random().nextInt(1000);
+        readRGSegmentToFile(segmentHelper, segmentStoreHost, scope, stream, tmpfilename, fileName);
+        output("The readerGroup stream has been successfully written into %s", fileName);
+    }
+
+    /**
+     * Reads the contents of the segment starting from the given offset and writes into the provided file.
+     *
+     * @param segmentHelper       A {@link SegmentHelper} instance to read the segment.
+     * @param segmentStoreHost    Address of the segment-store to read from.
+     * @param scope               The name of the scope.
+     * @param stream              The name of the stream.
+     * @param fileName            A name of the file to which the data will be written.
+     * @throws IOException if the file create/write fails.
+     * @throws Exception if the request fails.
+     */
+    private void readRGSegmentToFile(SegmentHelper segmentHelper, String segmentStoreHost, String scope, String stream,
+                                     String tmpfilename, String fileName) throws IOException, Exception {
+
+        File outputfile = FileHelper.createFileAndDirectory(fileName);
+
+        @Cleanup
+        BufferedWriter writer = new BufferedWriter(new FileWriter(outputfile));
+
+        Controller controller = instantiateController();
+        CompletableFuture<StreamSegments> streamSegments = controller.getCurrentSegments(scope, stream);
+        StreamSegments ss = streamSegments.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        int segmentid = (ss.getNumberOfSegments() > 0) ? (ss.getNumberOfSegments() - 1) : 0;
+        String fullyQualifiedSegmentName = NameUtils.getQualifiedStreamSegmentName(scope, stream, segmentid);
+
+        CompletableFuture<WireCommands.StreamSegmentInfo> segmentInfo = segmentHelper.getSegmentInfo(fullyQualifiedSegmentName,
+                            new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), authHelper.retrieveMasterToken(), 0L);
```

Comment:
Ok, so you are using the Admin Gateway to issue this command. Here you are also assuming that the user knows the Segment Store to which this request should be sent, right? Wondering if we could make it easier for a user by making this command to also lookup the segment store owning a given reader group segment.

Reply:
We need a new command to help the user know the endpoint for a Segment to help him/her running all these Wire Admin Commands. I have create a separate issue for that: https://github.com/pravega/pravega/issues/6485.